src/Symfony/Component/Messenger/Envelope.php

```diff
@@ -86,4 +86,15 @@ public function getMessage()
     {
         return $this->message;
     }
+
+    /**
+     * @param object $target
+     *
+     * @return Envelope|object The original message or the envelope if the target supports it
+     *                         (i.e implements {@link EnvelopeAwareInterface}).
+     */
+    public function getMessageFor($target)
```

Comment:
IMO, this should rather be a private method in TraceableMiddleware instead of exposing this method here publicly